File: 525-contiguous-array/525-contiguous-array.java
class Solution {
    public int findMaxLength(int[] nums) {
        Map<Integer, Integer> hashMap = new HashMap<>();
        hashMap.put(0, -1);
        int maxLength = 0, count = 0;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == 0){
                count = count - 1;
            }else{
                count = count + 1;
            }
            if(hashMap.containsKey(count)){
                maxLength = Math.max(maxLength, i - hashMap.get(count));
            }else{
                hashMap.put(count, i);
            }
        }
        return maxLength;
    }
}

File: Kashif/501-600/525-contiguous-array/525-contiguous-array.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        unordered_map<int, int> hashMap;
        hashMap[0] = -1;
        int maxLength = 0, count = 0;
        for(int i = 0; i< nums.size(); i++){
            if(nums[i] == 0){
                count--;
            }else{
                count++;
            }
            if(hashMap.find(count) != hashMap.end()){
                maxLength = max(maxLength, i - hashMap[count]);
            }else{
                hashMap[count] = i;
            }
        }
        return maxLength;
    }
};